### PR TITLE
[FLINK-36346][API] Remove deprecated API in flink-streaming-java module

### PR DESCRIPTION
**Pull request by reswqa**
## What is the purpose of the change

*Remove deprecated API in flink-streaming-java module*

## Brief change log

  - *Remove deprecated API in flink-streaming-java module*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by codenohup** [comment]
LGTM overall, I have left some comments, cloud you have a look? @reswqa

File: flink-table/flink-sql-gateway/src/test/resources/log4j2-test.properties [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 # Set root logger level to OFF to not flood build logs
 # set manually to INFO for debugging purposes
-rootLogger.level = OFF
+rootLogger.level = INFO
```
[the remark this answers is not in the document]
Comment (reswqa) [reply flink-table/flink-sql-gateway/src/test/resources/log4j2-test.properties]: Oops, unrelated change using for debug purpose. Good catch!

File: flink-table/flink-sql-gateway/pom.xml [hunk 2]
```diff
@@ -51,6 +51,12 @@
             <version>${project.version}</version>
         </dependency>
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
```
[the remark this answers is not in the document]
Comment (reswqa) [reply flink-table/flink-sql-gateway/pom.xml]: Should be removed.

File: flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/IntervalJoinITCase.java [hunk 3]
```diff
@@ -361,6 +362,7 @@ public void testBoundsAreInclusiveByDefault() throws Exception {
 
         DataStream<Tuple2<String, Integer>> streamOne =
                 env.fromData(Tuple2.of("key", 0), Tuple2.of("key", 1), Tuple2.of("key", 2))
+                        .assignTimestampsAndWatermarks(WatermarkStrategy.forMonotonousTimestamps())
```
[the remark this answers is not in the document]
Comment (reswqa) [reply flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/IntervalJoinITCase.java:365]: I believe only the second one is needed(The first one is for debugging forget to delete...).

File: flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/SourceNAryInputChainingITCase.java [hunk 4]
```diff
@@ -365,7 +365,13 @@ private static JobGraph sinkAndCompileJobGraph(DataStream<?> stream) {
         return StreamingJobGraphGenerator.createJobGraph(streamGraph);
     }
 
-    private static void verifySequence(final List<Long> sequence, final long from, final long to) {
+    private static void verifySequence(
+            final CloseableIterator<Long> iter, final long from, final long to) {
+        List<Long> sequence = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (reswqa) [reply flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/SourceNAryInputChainingITCase.java:370]: Yep, good suggestion.

**Comment by reswqa** [comment]
@codenohup Thanks for the careful review! I have updated this PR, PTAL again.

**Review by codenohup — APPROVED** [review]
LGTM, thank you for the update!

**Pull request merged**